File: code_blocks/integrations/third-party-integrations/statsig_10.dart
import 'package:statsig/statsig.dart';
import 'package:purchases_flutter/purchases_flutter.dart';

Future<void> setupRevenueCatWithStatsig() async {
  await Purchases.configure(
    PurchasesConfiguration('public_sdk_key')..appUserId = 'your_app_user_id',
  );

  await Statsig.initialize(
    'STATSIG_CLIENT_KEY',
    user: StatsigUser(userID: 'your_app_user_id'),
  );

  final stableID = Statsig.getStableID();
  await Purchases.setAttributes({'$statsigUserID': stableID});
}

File: code_blocks/integrations/third-party-integrations/statsig_10.ts
import { Statsig } from 'statsig-react-native';
import Purchases from '@revenuecat/purchases-react-native';

export async function setupRevenueCatWithStatsig() {
  await Purchases.configure({
    apiKey: 'public_sdk_key',
    appUserID: 'your_app_user_id',
  });

  await Statsig.initialize('STATSIG_CLIENT_KEY', { userID: 'your_app_user_id' });

  const stableID = Statsig.getStableID();
  await Purchases.setAttributes({ '$statsigUserID': stableID });
}
